controllers/workspace/devworkspace_controller.go

```diff
@@ -360,6 +465,34 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		devfilePodAdditions.InitContainers = append([]corev1.Container{*projectClone}, devfilePodAdditions.InitContainers...)
 	}
 
+	// Inject operator-configured init containers
+	if workspace.Config != nil && workspace.Config.Workspace != nil {
+		// Check if init-persistent-home should be disabled
+		disableHomeInit := workspace.Config.Workspace.PersistUserHome.DisableInitContainer != nil &&
+			*workspace.Config.Workspace.PersistUserHome.DisableInitContainer
+
+		for _, c := range workspace.Config.Workspace.InitContainers {
+			// Special handling for init-persistent-home
+			if c.Name == constants.HomeInitComponentName {
+				// Skip if persistent home is disabled
+				if !home.PersistUserHomeEnabled(workspace) {
+					continue
+				}
+				// Skip if init container is explicitly disabled
+				if disableHomeInit {
+					continue
+				}
+				// Apply defaults and validation for init-persistent-home
+				validated, err := defaultAndValidateHomeInitContainer(c, workspace)
```

Comment:
Not sure, if we can do in this way, but can you check?
```suggestion
				container, err = defaultAndValidateHomeInitContainer(c, workspace)
```